demisto_sdk/commands/common/content/objects/pack_objects/dashboard/dashboard.py

```diff
@@ -21,3 +30,165 @@ def upload(self, client: demisto_client):
             The result of the upload command from demisto_client
         """
         return client.import_dashboard(file=self.path)
+
+    def validate(self):
+        old_file = get_old_file(self.path, self.base.old_file_path, self.base.prev_ver)
+        return self.is_valid_dashboard(old_file)
+
+    def is_valid_dashboard(self, old_file):
+        # type: (dict) -> bool
+        """Check whether the dashboard is valid or not.
+
+        Returns:
+            bool. Whether the dashboard is valid or not
+        """
+        is_dashboard_valid = all([
+            self.is_valid_version(),
+            self.is_valid_fromversion()
+        ])
+
+        # check only on added files
+        if not old_file:
+            is_dashboard_valid = all([
+                is_dashboard_valid,
+                self.is_id_equals_name(),
+                self.is_including_fields(),
+                self.contains_forbidden_fields()
+            ])
+
+        return is_dashboard_valid
+
+    def is_valid_fromversion(self):
```

Comment:
not sure why we're implementing this in every validator class and not just in its parent, lets discuss